ambry-server/src/main/java/com/github/ambry/server/RepairRequestsDB.java

```diff
@@ -0,0 +1,268 @@
+  private static final String EXPIRATION_TYPE = "expirationTime";
+
+  // LOCAL_CONSISTENCY_TODO continue the Get from one token
+  /**
+   * Select the records for one partition with the oldest operation time.
+   */
+  // @formatter:off
+  private static final String GET_QUERY = String.format(""
+      + "SELECT %s, %s, %s, %s, %s, %s, %s, %s "
+      + "FROM %s "
+      + "WHERE %s = ? "
+      + "ORDER BY %s ASC "
+      + "LIMIT 100",
+      BLOB_ID, PARTITION_ID, SOURCE_HOST_NAME, SOURCE_HOST_PORT, OPERATION_TYPE, OPERATION_TIME, LIFE_VERSION, EXPIRATION_TYPE,
+      REPAIR_REQUESTS_TABLE,
+      PARTITION_ID,
+      OPERATION_TIME);
+  // @formatter:on
+
+  /**
+   * Attempt to insert a RepairRequestRecord to the database
+   */
+  // @formatter:off
+  private static final String INSERT_QUERY = String.format(""
+      + "INSERT INTO %1$s "
+      + "(%2$s, %3$s, %4$s, %5$s, %6$s, %7$s, %8$s, %9$s) VALUES (?, ?, ?, ?, ?, ?, ?, ?)",
+      REPAIR_REQUESTS_TABLE,
+      BLOB_ID, PARTITION_ID, SOURCE_HOST_NAME, SOURCE_HOST_PORT, OPERATION_TYPE, OPERATION_TIME, LIFE_VERSION, EXPIRATION_TYPE);
+  // @formatter:on
+
+  /**
+   * Delete a record from the database
+   */
+  // @formatter:off
+  private static final String DELETE_QUERY = String.format(""
+      + "DELETE FROM %s "
+      + "WHERE %s=? AND %s=?",
+      REPAIR_REQUESTS_TABLE,
+      BLOB_ID, OPERATION_TYPE);
+  // @formatter:on
+
+  private final String localDatacenter;
+  private final DataSource dataSource;
+  private final ServerMetrics metrics;
+  private final MySqlUtils.DbEndpoint dbEndpoint;
+  private final Connection connection;
+
+  public RepairRequestsDB(ServerConfig config, ServerMetrics metrics, String localDatacenter) throws SQLException {
+    // Get the db endpoint for this local data center.
+    if (config.dbInfo == null || config.dbInfo.isEmpty()) {
+      logger.warn("RepairRequestsDB: didn't specify the db information.");
+      throw new SQLException("RepairRequestsDB: didn't specify the db information.");
+    }
+    List<MySqlUtils.DbEndpoint> dbs = MySqlUtils.getDbEndpointsPerDC(config.dbInfo)
+        .values()
+        .stream()
+        .flatMap(List::stream)
+        .filter(MySqlUtils.DbEndpoint::isWriteable)
+        .filter(endpoint -> localDatacenter.equals(endpoint.getDatacenter()))
+        .collect(Collectors.toList());
+    if (dbs == null || dbs.size() == 0) {
+      logger.warn("RepairRequestsDB: doesn't have the valid db. " + config.dbInfo);
+      throw new SQLException("RepairRequestsDB: doesn't have the valid db. " + config.dbInfo);
+    }
+
+    // connect to the database
+    dbEndpoint = dbs.get(0);
+    dataSource = buildDataSource(dbEndpoint);
+    this.localDatacenter = localDatacenter;
+    this.metrics = metrics;
+
+    Properties credentials = new Properties();
+    credentials.setProperty("user", dbEndpoint.getUsername());
+    credentials.setProperty("password", dbEndpoint.getPassword());
```

Comment:
I recommend that you just use dataSource to get connections, like named blob db.

Reply:
Fixed.